src/component.cpp

```diff
@@ -142,12 +142,12 @@ std::string Component::doSerialisation(FORMATS format) const
         std::string componentName = getName();
         if (componentName.length()) {
             repr += " name=\"" + componentName + "\"";
-            if (variableCount() > 0) {
-                endTag = true;
-                repr += ">";
-                for(size_t i = 0; i < variableCount(); i++) {
-                    repr += getVariable(i)->serialise(format);
-                }
+        }
+        if (variableCount() > 0) {
+            endTag = true;
+            repr += ">";
+            for(size_t i = 0; i < variableCount(); i++) {
```

Comment:
also, I'd suggest it might be best to resolve the current set of pull requests and then have a concerted "code tidy up" - for the code outside the direct scope of this chunk of work. For these specific instances it is best to fix them here and now.